### PR TITLE
(4.x) Merge 3.4

### PR DESCRIPTION
**Pull request by alalek**
#20200 from rmeertens:update_grabcut
#20394 from SamFC10:conv-asymmetric-pads
#20397 from alalek:fix_winpack_dldt_stale_sysroot
#20399 from pablorcum:3.4
#20410 from alalek:fix_dnn_dldt_output_layout
#20420 from berak:python_fix_trackbar_warning

Previous "Merge 3.4": #20387

<cut/>

<details>

```
Xci_branch=merge-3.4
buildworker:Win64 OpenCL=windows-2
#buildworker:Custom=linux-1,linux-2,linux-4,linux-6
buildworker:Docs=linux-4,linux-6
build_image:Docs=docs-js:18.04
Xbuild_image:Custom=javascript
Xbuildworker:Custom=linux-4,linux-6
Xbuild_image:Custom=javascript-simd
#build_image:Custom=powerpc64le
#build_image:Custom=ubuntu-openvino-2019r3.0:16.04
#build_image:Custom=ubuntu-openvino-2020.3.0:16.04
#build_image:Custom=ubuntu-openvino-2020.4.0:16.04
#build_image:Custom=ubuntu-openvino-2021.1.0:20.04
#build_image:Custom=ubuntu-openvino-2021.2.0:20.04
#build_image:Custom=ubuntu-openvino-2021.3.0:20.04
build_image:Custom=ubuntu-openvino-2021.4.0:20.04
#buildworker:Custom=linux-1
#build_image:Custom=ubuntu-vulkan:16.04
#buildworker:Custom=linux-4
#build_image:Custom=fedora:28
#build_image:Custom=ubuntu-cuda:16.04
#build_image:Custom=ubuntu-clang:18.04
#build_image:Custom=ubuntu:20.04
#buildworker:Custom=linux-1
#build_image:Custom=javascript-simd
#build_image:Custom=mips64el
#build_image:Custom Mac=openvino-2019r3.0
#build_image:Custom Mac=openvino-2020.3.0
Xbuild_image:Custom Mac=openvino-2020.4.0
Xbuild_image:Custom Mac=openvino-2021.1.0
Xbuild_image:Custom Mac=openvino-2021.2.0
Xbuild_image:Custom Mac=openvino-2021.3.0
build_image:Custom Mac=openvino-2021.4.0
#build_image:Custom Win=openvino-2019r3.0
#build_image:Custom Win=openvino-2020.3.0
Xbuild_image:Custom Win=openvino-2020.4.0
Xbuild_image:Custom Win=openvino-2021.1.0
Xbuild_image:Custom Win=openvino-2021.2.0
Xbuild_image:Custom Win=openvino-2021.3.0
Xbuild_image:Custom Win=openvino-2021.4.0
test_bigdata:Custom Win=1
test_filter:Custom Win=*OpenVINO*:*onv*
test_modules=dnn,gapi,python2,python3,java
test_opencl:Custom Win=ON
build_contrib:Custom Win=OFF
#build_image:Custom Win=msvs2017
#build_image:Custom Win=msvs2019
test_modules:Custom Mac=dnn,gapi,java,python3

build_image:Custom Win=winpack-dldt

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=ON
test_bigdata:Custom=1
test_filter:Custom=*OpenVINO*:*onv*
```

</details>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -659,7 +659,7 @@ if(UNIX)
     elseif(EMSCRIPTEN)
       # no need to link to system libs with emscripten
     elseif(QNXNTO)
-      # no need to link to system libs with QNX
+      set(OPENCV_LINKER_LIBS ${OPENCV_LINKER_LIBS} m)
```
Comment (alalek) [inline CMakeLists.txt:662]: @pablorcum @blackliner Could you please check merged QNX-related changes? (PR #20399)

Comment (blackliner) [reply CMakeLists.txt:662]: LGTM: https://www.qnx.com/developers/docs/6.4.1/neutrino/lib_ref/c/cos.html

**Review by blackliner — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (pablorcum) [reply CMakeLists.txt:662]: I've built it successfully for QNX after this change.

**Review by pablorcum — APPROVED** [review]

**Comment by alalek** [comment]
:+1:

**Pull request merged**